isovar/translation.py

```diff
@@ -42,19 +42,53 @@ class Translation(TranslationKey):
     def __init__(
             self,
             amino_acids,
-            variant_aa_interval_start,
-            variant_aa_interval_end,
+            contains_mutation,
+            mutation_start_idx,
+            mutation_end_idx,
             ends_with_stop_codon,
             frameshift,
             untrimmed_variant_sequence,
             reference_context,
             variant_orf):
+        """
+        Parameters
+        ----------
+        amino_acids : str
+            Amino acid sequence
+
+        contains_mutation : bool
```

Comment:
general impression is that the detailed logic around start/end indices, whether something is a frameshift or deletion/etc. seems to be spread across a number of places. It's not easy to follow, but maybe this is unavoidable?

Reply:
Yeah, I also felt like it's too spread out but didn't want to refactor it more / change things to streamline that logic.